clients/drcachesim/tracer/raw2trace.h

```diff
@@ -740,11 +755,21 @@ template <typename T> class trace_converter_t {
         }
         DR_ASSERT(in_entry->tid.type == OFFLINE_TYPE_THREAD);
         header->tid = in_entry->tid.tid;
+
         in_entry = impl()->get_next_entry(tls);
         if (in_entry == nullptr)
             return "Failed to read header from input file";
         DR_ASSERT(in_entry->pid.type == OFFLINE_TYPE_PID);
         header->pid = in_entry->pid.pid;
+
+        in_entry = impl()->get_next_entry(tls);
+        if (in_entry == nullptr)
+            return "Failed to read header from input file";
```

Comment:
This should handle compability issues.  I.e., it should handle a trace that does not have this new marker.  If the new marker is not there, it should unread_last_entry and maybe use an estimate (better than 0) for the line size, like the current processor's line size.  See below as well.

Reply:
The header processing code in raw2trace.cpp now sets the current processor cache line size as default if the header doesn't contain it.

Reply:
Do we need a dedicated offline trace analyser test to verify this fallback? I added a log at level-2 and verified locally the fallback path gets invoked.

Reply:
Another option would be to bump the trace file version and use that to see for sure whether the cache line size should be there but it doesn't seem necessary to go that far.

IMHO a manual test for the missing entry is enough.

Reply:
Thanks. Done.